src/app/clusters/microwave-oven-control-server/microwave-oven-control-server.cpp

```diff
@@ -0,0 +1,166 @@
+    {
+        ChipLogError(Zcl, "Microwave Oven Control: The cluster with ID %lu was not enabled in zap.", long(mClusterId));
+        return CHIP_ERROR_INVALID_ARGUMENT;
+    }
+
+    ReturnErrorOnFailure(InteractionModelEngine::GetInstance()->RegisterCommandHandler(this));
+    VerifyOrReturnError(registerAttributeAccessOverride(this), CHIP_ERROR_INCORRECT_STATE);
+    return CHIP_NO_ERROR;
+}
+
+uint32_t Instance::GetCookTime() const
+{
+    return mCookTime;
+}
+
+uint8_t Instance::GetPowerSetting() const
+{
+    return mPowerSettng;
+}
+
+void Instance::SetCookTime(uint32_t cookTime)
+{
+    mCookTime = cookTime;
+}
+
+void Instance::SetPowerSetting(uint8_t powerSetting)
+{
+    mPowerSettng = powerSetting;
+}
+
+CHIP_ERROR Instance::Read(const ConcreteReadAttributePath & aPath, AttributeValueEncoder & aEncoder)
+{
+    ChipLogError(Zcl, "OperationalState: Reading");
+    switch (aPath.mAttributeId)
+    {
+    case MicrowaveOvenControl::Attributes::CookTime::Id: {
+        ReturnErrorOnFailure(aEncoder.Encode(GetCookTime()));
+    }
+    break;
+
+    case MicrowaveOvenControl::Attributes::PowerSetting::Id: {
+        ReturnErrorOnFailure(aEncoder.Encode(GetPowerSetting()));
+    }
+    break;
+
+    case MicrowaveOvenControl::Attributes::MinPower::Id: {
+        ReturnErrorOnFailure(aEncoder.Encode(mDelegate->GetMinPower()));
+    }
+    break;
+
+    case MicrowaveOvenControl::Attributes::MaxPower::Id: {
+        ReturnErrorOnFailure(aEncoder.Encode(mDelegate->GetMaxPower()));
+    }
+    break;
+
+    case MicrowaveOvenControl::Attributes::PowerStep::Id: {
+        ReturnErrorOnFailure(aEncoder.Encode(mDelegate->GetPowerStep()));
+    }
+    break;
+    }
+    return CHIP_NO_ERROR;
+}
+
+void Instance::InvokeCommand(HandlerContext & handlerContext)
+{
+    ChipLogDetail(Zcl, "MicrowaveOvenControl: InvokeCommand");
+    switch (handlerContext.mRequestPath.mCommandId)
+    {
+    case Commands::SetCookingParameters::Id:
+        ChipLogDetail(Zcl, "MicrowaveOvenControl:  Entering SetCookingParameters");
+
+        CommandHandlerInterface::HandleCommand<Commands::SetCookingParameters::DecodableType>(
+            handlerContext, [this](HandlerContext & ctx, const auto & req) { HandleSetCookingParameters(ctx, req); });
+        break;
+
+    case Commands::AddMoreTime::Id:
+        ChipLogDetail(Zcl, "MicrowaveOvenControl:  Entering AddMoreTime");
+
+        CommandHandlerInterface::HandleCommand<Commands::AddMoreTime::DecodableType>(
+            handlerContext, [this](HandlerContext & ctx, const auto & req) { HandleAddMoreTime(ctx, req); });
+        break;
+    }
+}
+
+void Instance::HandleSetCookingParameters(HandlerContext & ctx, const Commands::SetCookingParameters::DecodableType & req)
+{
+    ChipLogDetail(Zcl, "MicrowaveOvenControl:  HandleSetCookingParameters");
+    Status status;
+    status = mDelegate->HandleSetCookingParametersCallback(req.cookMode, req.cookTime, req.powerSetting);
```

Comment:
Why is this this not doing the spec-required constraint validations (see Constraint) column in the spec?

Why is this not doing the spec-required Operational State validations?